CONTRIBUTING.md

```diff
@@ -30,44 +30,76 @@ Improvements to documentation are always appreciated! If you see areas that coul
-
-3. To build all services, run:
-   ```
-   docker buildx bake --file docker-bake.hcl
-   ```
-
-4. To build a specific service, use:
-   ```
-   docker buildx bake --file docker-bake.hcl <service-name>
-   ```
-   Replace `<service-name>` with one of the following:
-   - agents-api
-   - agents-api-worker
-   - cozo-migrate
-   - memory-store
-   - integrations
-   - gateway
-   - embedding-service-cpu
-   - embedding-service-gpu
-
-5. To set a custom tag for the images, use:
-   ```
-   docker buildx bake --file docker-bake.hcl --set *.tags=myorg/myimage:v1.0
-   ```
-   Replace `myorg/myimage:v1.0` with your desired image name and tag.
-
-6. By default, the images are built with the "latest" tag. To specify a different tag, you can set the TAG variable:
-   ```
-   docker buildx bake --file docker-bake.hcl --set TAG=v1.2.3
-   ```
-
-Note: The `docker-bake.hcl` file defines the build contexts, Dockerfiles, and tags for each service. If you need to modify the build process for a specific service, update the corresponding target in the `docker-bake.hcl` file.
-
-Thank you for your interest in contributing to this project!
\ No newline at end of file
+### Setup Instructions
+
+##### 1. Clone the Repository
+Clone the repository from your preferred source:
+
+```bash
+git clone <repository_url>
+```
+
+##### 2. Navigate to the Root Directory
+Change to the root directory of the project:
+
+```bash
+cd <repository_root>
+```
+
+##### 3. Set Up Environment Variables
+- Create a `.env` file in the root directory.
+- Refer to the `.env.example` file for a list of required variables.
+- Ensure that all necessary variables are set in the `.env` file.
+
+##### 4. Create a Docker Volume for Backup
+Create a Docker volume named `cozo_backup`:
+
+```bash
+docker volume create cozo_backup
+```
+
+##### 5. Run the Project using Docker Compose
+You can run the project in two different modes: **Single Tenant** or **Multi-Tenant**. Choose one of the following commands based on your requirement:
+
+###### Single-Tenant Mode
+Run the project in single-tenant mode:
+
+```bash
+docker compose --env-file .env --profile temporal-ui --profile single-tenant --profile embedding-cpu --profile self-hosted-db up --force-recreate --build --watch
+```
+
+> **Note:** In single-tenant mode, you can interact with the SDK directly without the need fopr the API KEY.
+
+###### Multi-Tenant Mode
+Run the project in multi-tenant mode:
+
+```bash
+docker compose --env-file .env --profile temporal-ui --profile multi-tenant --profile embedding-cpu --profile self-hosted-db up --force-recreate --build --watch
+```
+
+> **Note:** In multi-tenant mode, you need to generate a JWT token that act as an API KEY to interact with the SDK in the local setup.
+
+##### 6. Generate a JWT Token (Only for Multi-Tenant Mode)
+
+To generate a JWT token, `jwt-cli` is required. Install it using the following command:
+
+```bash
+npm install -g jwt-cli
```

Comment:
no the lib is different: `cargo install jwt-cli --locked`

Reply:
for cargo installation the users might require extra set installation like Rust and Cargo. Instead can we just ask to install 'jwt-cli' depending on their platform?